### PR TITLE
Add tool to convert WKT geometries to a TGF graph

### LINKED ISSUE #62: Tool to convert connected geometries into a graph
Necessary to implement #49, #51, and maybe #50.

Will probably be easier (or perhaps just more performant?) after implementing #6.

- [x] Should output undirected graph in [TGF](https://en.wikipedia.org/wiki/Trivial_Graph_Format) with each node labeled with its POINTZ coordinates
- [x] Should be able to consume any geometries
  - [x] disjoint points shall be discarded, e.g., POINT(0 0) + LINESTRING(1 1, 2 2) -> LINESTRING(1 1, 2 2)
  - [x] points on an edge shall be discarded, e.g., POINT(1 1) + LINESTRING(0 0, 2 2) -> LINESTRING(0 0, 2 2)
  - [x] Geometry type is not stored as metadata, so you lose individual geometry distinctness
- [ ] (future) Should be able to convert the same graph format back into LINESTRINGs (no other types)
- [ ] Graph should be undirected with no duplicate edges
  - [ ] (optional) multiple collinear edges should be collapsed into a single edge when possible (perhaps leave for simplifying in a separate tool?)
- [x] Nodes within some tolerance of each other should be snapped together
- [ ] (optional) Split noding into its own tool?
- [ ] (optional) Use google benchmark to profile graph generation

### PR DESCRIPTION
**Pull request by Notgnoshi**
Closes #62 

TODO items (to be done before merge):
- [x] More substantive testing
  - [x] Test for duplicate edges
  - [x] Edge cases (pun intended)
  - [x] Count nodes/edges
- [x] Visualization
- [x] Refactor graph output
  - [x] Pick XML, DOT, or TGF? (probably just skip XML...)
  - [x] Opportunity to apply some design patterns :wink:

TODO items (not to be done in this PR):
- [ ] Convert from graph to MULTILINESTRING
- [ ] Collapse collinear edges joined by vertices of degree 2 into a single edge. Perhaps perform DP simplification?
- [ ] Benchmark and performance testing
- [ ] Decide whether to force 3D output (maybe mixed dimensionality is a feature)
- [ ] WKB (hex) input?

### CODE REVIEW

**Comment by Notgnoshi** [comment]
When printing, only print an edge for the lower of the two node ids.

File: tools/geom2graph/src/geom2graph/geometry-flattener.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            *m_geometry.getGeometryN(m_index), 0);
+    }
+
+    return *this;
+}
+
+bool GeometryFlattener::RecursiveGeometryIterator::operator==(
+    const RecursiveGeometryIterator& rhs) const
+{
+    return (&this->m_geometry == &rhs.m_geometry) && this->m_index == rhs.m_index;
```
Comment (Notgnoshi) [inline tools/geom2graph/src/geom2graph/geometry-flattener.cpp:129]: This should recursively call m_iterator->operator==, if m_iterator is not null.

File: tools/geom2graph/include/geom2graph/noding/geometry-graph.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace geom2graph::noding {
+class GeometryGraph
+{
+public:
+    struct Node
+    {
+        const std::size_t id;  //!< An index into the GeometryGraph::get_graph() array.
+        const geos::geom::Coordinate coord;  //!< The coordinate for this node in the graph.
+        std::set<std::size_t> adjacencies;   //!< All of the Node id's adjacent to this node.
```
Comment (Notgnoshi) [inline tools/geom2graph/include/geom2graph/noding/geometry-graph.h:19]: This should be an unordered_set
